### PR TITLE
docs: debugging doc update

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

First part of debugging docs refresh.

Next steps is to add more info about break points and other debuggers. This one mostly adds flipper how-to and installation steps.

Happy to hear your thoughts

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- debugging doc update

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: docs/debugging.md [hunk 1]
```diff
@@ -1,23 +1,10 @@
...
-   1a. Developer mode should be on in you are developing (`__DEV__` is true, or you are logged in with an artsy email). Make sure it is, by going to Profile > About. If "Version" has a one pixel purple line on the right, then you are good to go. If not, then you need to tap "Version" 7 times quickly, to enable developer mode.
-2. Simulate a ‘shake’ event (<kbd>^⌘Z</kbd>), which will bring up the Dev Menu.
-
-## In-App Developer Menu
-
-You can access the developer menu by using the <kbd>⌘D</kbd> keyboard shortcut when your app is running in the iOS Simulator, or <kbd>⌘M</kbd> when running in an Android emulator on Mac OS and Ctrl+M on Windows and Linux.
-![In-app-developer-menu](./screenshots/in-app-developer-menu.png)
+- [React Native Debugger](https://github.com/jhen0409/react-native-debugger)
+- [Flipper](https://fbflipper.com/)
+- [VSCode](https://blog.logrocket.com/debugging-react-native-vs-code/)
```
Comment (damassi) [inline docs/debugging.md:7]: Should we add a link to the dev menu doc from here, somewhere close to the top?

**Review by dimatretyak — APPROVED** [review]
Thank you very much 🙏

**Review by MrSltun — APPROVED** [review]
✨✨✨

**Comment by damassi** [comment]
docs look amazing, tyy

**Pull request merged**